packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroup.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import { MoreHorizontalRegular } from '@fluentui/react-icons';
+import { PopoverSurface } from '@fluentui/react-popover';
+import type { AvatarGroupProps, AvatarGroupState } from './AvatarGroup.types';
+
+/**
+ * Create the state required to render AvatarGroup.
+ *
+ * The returned state can be modified with hooks such as useAvatarGroupStyles_unstable,
+ * before being passed to renderAvatarGroup_unstable.
+ *
+ * @param props - props from this instance of AvatarGroup
+ * @param ref - reference to root HTMLElement of AvatarGroup
+ */
+export const useAvatarGroup_unstable = (props: AvatarGroupProps, ref: React.Ref<HTMLElement>): AvatarGroupState => {
+  const { children, layout = 'spread', maxAvatars = 5, overflowIndicator = 'count', size = 32 } = props;
+  const childrenArray = React.Children.toArray(children);
+  const childrenCount = childrenArray.length;
+  const numOfAvatarsToShow = layout === 'pie' ? 3 : maxAvatars;
+  const hasOverflow = childrenCount > numOfAvatarsToShow;
+  const numOfAvatarsToHide = childrenCount - numOfAvatarsToShow + 1;
+
+  let rootChildren = childrenArray;
+  let popoverChildren = null;
+  let popoverTriggerChildren = null;
```

Comment:
(super-nit) prefer `undefined` instead of `null` when possible. And since `undefined` is the default value of a `let`, this could be:
```suggestion
  let popoverChildren;
  let popoverTriggerChildren;
```